docs/ConnectAuthorizationAttachTypes.md

```diff
@@ -0,0 +1,384 @@
+2. **Tunnel-aware security** - Different policies for tunneled vs. non-tunneled traffic
+3. **Route-dependent access control** - Authorization decisions that depend on the network path
+
+## Solution: CONNECT_AUTHORIZATION Attach Types
+
+The new `BPF_CGROUP_INET4_CONNECT_AUTHORIZATION` and `BPF_CGROUP_INET6_CONNECT_AUTHORIZATION` attach types address this limitation by operating at the **authorization layer** in WFP.
+
+### Key Characteristics
+
+- **Timing**: Invoked **after route selection** but **before outbound connection authorization**
+- **Available Information**: Full outbound connection context including interface and tunnel information
+- **Purpose**: Authorization decisions based on complete outbound connection metadata
+- **Limitation**: **No redirection support** - outbound connections cannot be redirected at this layer since route selection has already occurred
+
+### WFP Layer Mapping
+
+- `BPF_CGROUP_INET4_CONNECT_AUTHORIZATION` → `FWPM_LAYER_ALE_AUTH_CONNECT_V4`
+- `BPF_CGROUP_INET6_CONNECT_AUTHORIZATION` → `FWPM_LAYER_ALE_AUTH_CONNECT_V6`
+
+## When to Use Each Attach Type
+
+### Use CONNECT Attach Types When:
+- You need to **redirect** outbound connections to different destinations (**redirection is only supported at these layers**)
+- You only need basic outbound connection information (source/destination IP/port)
+- Interface and route information is not relevant to your use case
+
+### Use CONNECT_AUTHORIZATION Attach Types When:
+- You need **interface information** for authorization decisions (via `bpf_sock_addr_get_interface_type()`)
+- You need **tunnel type** information (via `bpf_sock_addr_get_tunnel_type()`)
+- You need **next-hop interface** details (via `bpf_sock_addr_get_next_hop_interface_luid()`)
+- You need **sub-interface** granularity (via `bpf_sock_addr_get_sub_interface_index()`)
+- You want to **authorize or deny** outbound connections based on complete network context (**no redirection support**)
+- Your policy depends on route-dependent metadata
+
+## Implementation Details
+
+### Program Type
+Both CONNECT_AUTHORIZATION attach types use the `BPF_PROG_TYPE_CGROUP_SOCK_ADDR` program type, sharing the same context structure and helper functions as other sock_addr programs.
+
+### Verdict Handling
+CONNECT_AUTHORIZATION programs can return:
+- `BPF_SOCK_ADDR_VERDICT_PROCEED_HARD` - Allow the outbound connection and skip further authorization checks
+- `BPF_SOCK_ADDR_VERDICT_PROCEED_SOFT` - Allow the outbound connection but continue with additional authorization checks
+- `BPF_SOCK_ADDR_VERDICT_REJECT` - Deny the outbound connection
+
+#### Verdict Types Explained
+
+**PROCEED_HARD vs PROCEED_SOFT:**
+- `PROCEED_HARD` provides an optimization by signaling to the WFP layer that no further authorization checks are needed for this outbound connection, potentially improving performance for trusted outbound connections
```

Comment:
a couple of points:
- PROCEED_HARD will tell WFP it does not need to verify any more filters **at this sublayer**. it will still process the outbound request at other sublayers.
- PROCEED_HARD will override a soft-block (e.g., default Windows Firewall allow/block settings are soft-block settings; this will allow the connection even if the Windows Firewall policy is to block by default)